service/header/interface.go

```diff
@@ -0,0 +1,39 @@
+package header
+
+import (
+	"context"
+
+	tmbytes "github.com/celestiaorg/celestia-core/libs/bytes"
+)
+
+// Subscriber encompasses the behavior necessary to
+// subscribe/unsubscribe from new ExtendedHeader events from the
+// network.
+type Subscriber interface {
+	Subscribe() (Subscription, error)
+}
+
+// Subscription can retrieve the next ExtendedHeader from the
+// network.
+type Subscription interface {
+	NextHeader(ctx context.Context) (*ExtendedHeader, error)
+	Cancel()
+}
+
+// Exchange encompasses the behavior necessary to request ExtendedHeaders
+// from the network.
+type Exchange interface {
+	RequestHeader(ctx context.Context, height int64) (*ExtendedHeader, error)
+	RequestBatchedHeaders(ctx context.Context, from, to int64) ([]*ExtendedHeader, error)
```

Comment:
Hmmm. Nit, but would `RequestHeaderBatch` be better? Or just `RequestHeaders`?

Reply:
RequestHeaders is fine

Reply:
Here is important to note that requested Headers here are not verified.
Also, +1 for `RequestHeaders`. The lesser amount the words - the better(surely preserving the same meaning)